tephi/tests/test_tephigram.py

```diff
@@ -60,164 +60,172 @@ def test_load_multiple_files_same_column_names(self):
         dews, temps = tephi.loadtxt(
             self.filename_dews, self.filename_temps, column_titles=columns
         )
-        self.assertEqual(dews._fields, columns)
-        self.assertEqual(temps._fields, columns)
+        assert dews._fields == columns
+        assert temps._fields == columns
 
     def test_load_data_too_many_column_iterables(self):
         columns = [
             ("pressure", "dewpoint"),
             ("pressure", "wind_speed", "wind_direction"),
         ]
-        with self.assertRaises(ValueError):
+        with pytest.raises(ValueError):
             tephi.loadtxt(self.filename_dews, column_titles=columns)
 
     def test_number_of_columns_and_titles_not_equal(self):
         columns = ("pressure", "dewpoint", "wind_speed")
-        with self.assertRaises(TypeError):
+        with pytest.raises(TypeError):
             tephi.loadtxt(self.filename_barbs, column_titles=columns)
 
     def test_invalid_column_titles(self):
         columns = ("pres-sure", "dew+point", 5)
-        with self.assertRaises(ValueError):
+        with pytest.raises(ValueError):
             tephi.loadtxt(self.filename_dews, column_titles=columns)
 
     def test_non_iterable_column_title(self):
         # For the case of column titles, strings are considered non-iterable.
         columns = "pressure"
-        with self.assertRaises(TypeError):
+        with pytest.raises(TypeError):
             tephi.loadtxt(self.filename_dews, column_titles=columns)
 
     def test_delimiter(self):
         columns = ("pressure", "temperature", "wind_direction", "wind_speed")
         data = tephi.loadtxt(
             self.filename_comma, column_titles=columns, delimiter=","
         )
-        self.assertEqual(data.pressure.shape, (2,))
+        assert data.pressure.shape == (2,)
 
     def test_dtype(self):
         dews = tephi.loadtxt(self.filename_dews, dtype="i4")
-        self.assertIsInstance(dews.pressure[0], np.int32)
-        self.assertIsInstance(dews.temperature[0], np.int32)
+        assert dews.pressure[0].dtype == np.int32
+        assert dews.temperature[0].dtype == np.int32
 
 
+@pytest.mark.graphical
+@pytest.mark.usefixtures("close_plot", "nodeid")
 class TestTephigramPlot(tests.GraphicsTest):
```

Comment:
@trexfeathers perhaps a utility test function may help here, nice suggestion, i'll give that a try

Reply:
@trexfeathers Okay, so I experimented with a utility method that attempted to perform the common actions of:
- creating the tephigram,
- customising the plot/s, and
- performing the final `check_graphic`. 

To be honest, the final result actually makes the tests look way more obfuscated and terse, forcing the effort on the reader to work out what the heck is going on and the purpose of the test.

So that said, I'd rather keep this obvious and simple. We've used a deep level of mixins for `iris` testing and although they have utility, the final result can quickly become cryptic and impenetrable.

So I rather just keep things explicit and simple here. Are you happy with that?

Reply:
On the readability vs D.R.Y. spectrum I seem to often be DRY-er than others. This probably isn't the place for me to start the debate. Let's leave it.

Not sure if this kind of this is captured in a style guide anywhere, or whether it should be.